lib/seahorn/BvOpSem2Context.hh

```diff
@@ -110,6 +112,11 @@ public:
   ~Bv2OpSemContext() override = default;
 
   EZ3 *getZ3() const { return m_z3.get(); }
+  std::shared_ptr<ZSimplifier<EZ3>> getSimplifier() { 
```

Comment:
Most of the time you would want to return `ZSimplifier<EZ3> &` 

What is the reason for sharing ownership with the caller?

Reply:
Fixed... may i know the reason for making it a shared_ptr in the first place... who is an entity we want to share the simplifier with?

Reply:
Not really sure. Maybe there was some issue with copy constructor or something like that. We can try moving it to `std::unique_ptr` instead.